src/main/java/Deadline.java

```diff
@@ -0,0 +1,46 @@
+public class Deadline extends Task {
+    private final String by;
+
+    public Deadline(String taskName, String by, boolean isCompleted) {
+        super(taskName, isCompleted);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.by + ")";
+    }
+
+    @Override
+    public String getStringRepresentation() {
+        return "deadline " + taskName + " /by " + this.by + getIsCompletedString();
+    }
+
+    public static Deadline getTask (String currentInput)
+        throws EmptyTaskDescription, InvalidTaskArguments {
+        try {
+            boolean isCompleted = currentInput.contains(Task.IS_COMPLETED_STRING);
+            currentInput = currentInput.replaceAll(Task.IS_COMPLETED_STRING, "");
+
+            int idxOfDeadline = currentInput.indexOf(" /by ");
+            if (idxOfDeadline == -1) {
+                throw new InvalidTaskArguments();
+            }
+            // Extract after _deadline_, which is 8 characters long
+            String taskName = currentInput.substring(8, idxOfDeadline);
```

Comment:
Avoid magic numbers like this, evident why 8 is used from comments, but maybe one other way to do it is to assign a variable AFTER_DEADLINE_POS to 8 so the code "self documents".